### ISSUE TITLE
args: Warning when local module use a local `module_utils`

### ISSUE THREAD

**Issue by zigarn** [ansible/ansible-lint#2835]
##### Summary

Warning when local module use a local `module_utils`.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
ansible-lint --version
ansible-lint 6.10.0 using ansible 2.12.4
```

##### STEPS TO REPRODUCE

```console
git clone https://opendev.org/zuul/zuul-jobs
cd zuul-jobs
ansible-lint
```

##### Desired Behavior

No error about importing local `module_utils`

##### Actual Behavior

```
WARNING  Ignored exception from ArgsRule.<bound method AnsibleLintRule.matchtasks of args: Validating module arguments.> while processing roles/use-buildset-registry/tasks/main.yaml (tasks): cannot import name 'remarshal' from 'ansible.module_utils' (/usr/lib/python3/dist-packages/ansible/module_utils/__init__.py)
```

**Comment by zigarn** [comment]
Created a quick and dirty fix:

```diff
diff --git i/src/ansiblelint/rules/args.py w/src/ansiblelint/rules/args.py
index 87c0aa4a..e5dc071d 100644
--- i/src/ansiblelint/rules/args.py
+++ w/src/ansiblelint/rules/args.py
@@ -6,6 +6,8 @@ import importlib.util
 import io
 import json
 import logging
+import os
+import pkgutil
 import re
 import sys
 from functools import lru_cache
@@ -33,6 +35,10 @@ _logger = logging.getLogger(__name__)
 @lru_cache
 def load_module(module_name: str) -> loader.PluginLoadContext:
     """Load plugin from module name and cache it."""
+    module_utils_paths = [p.path for p in loader.module_utils_loader._get_paths_with_context(subdirs=False) if not p.internal and os.path.isdir(p.path)]
+    for pkg in pkgutil.walk_packages(module_utils_paths):
+        module = pkg.module_finder.find_module(pkg.name).load_module(pkg.name)
+        sys.modules['ansible.module_utils.' + pkg.name] = module
     return loader.module_loader.find_plugin_with_context(module_name)
```

With this, local `module_utils.*` are loaded inside `ansible.module_utils` and modules importing them can work.

But not sure it's an elegant solution, and for sure it has to be somewhere outside this function and called only once.

**Comment by ssbarnea** [comment]
@ganeshrn WDYT about this? Does it look legit to you? The big question is if that is matching ansible-core behavior as we do not want to accidentally load a module that ansible itself will fail to load in production.

**Comment by zigarn** [comment]
From what I understood of ansible code: when building the ansiballz script the module's imports are analyzed and custom module_utils are kind of statically imported in the ansiballz.

**Comment by ssbarnea** [comment]
@zigarn I am going to close this as wontfix because I was not able to reproduce the bug with module from within a collection, this being the only one we support. In fact I was not able to get the error even with zuul-jobs repo.

If you can find a collection published on galaxy hub that can reproduce this bug, we will be more than happy to accept a pull request.

I even adapted your patch and run it but I need a reproducer in order to accept a patch, especially one like this. Ideally it would be a small example collection to reproduce the bug and the fix.

**Comment by zigarn** [comment]
Issue may have been fixed somehow another way.
Will see if I can still reproduce when I'll have time.

**Comment by damienrg** [comment]
Here is a reproducer with https://github.com/containers/ansible-podman-collections/:

```yaml
- name: Using Podman collection
  hosts: localhost
  tasks:
    - name: Trigger a warning with ansible-lint
      containers.podman.podman_save:
        name: nginx
```
